tests/test_coef.py

```diff
@@ -0,0 +1,446 @@
+import numpy as np
+import pytest
+
+from aspire.basis import (
+    Coef,
+    FBBasis2D,
+    FFBBasis2D,
+    FLEBasis2D,
+    FPSWFBasis2D,
+    PSWFBasis2D,
+)
+from aspire.utils import utest_tolerance
+
+IMG_SIZE = [
+    31,
+    32,
+]
+DTYPES = [
+    np.float32,
+    np.float64,
+]
+STACKS = [
+    (),
+    (1,),
+    (2,),
+    (3, 4),
+]
+
+ALLYOURBASES = [
+    FBBasis2D,
+    FFBBasis2D,
+    PSWFBasis2D,
+    FPSWFBasis2D,
+    FLEBasis2D,
+]
+
+
+def sim_fixture_id(params):
+    stack, count, dtype = params
+    return f"stack={stack}, count={count}, dtype={dtype}"
+
+
+@pytest.fixture(params=DTYPES, ids=lambda x: f"dtype={x}", scope="module")
+def dtype(request):
+    """
+    Dtypes for coef array
+    """
+    return request.param
+
+
+@pytest.fixture(params=DTYPES, ids=lambda x: f"dtype={x}", scope="module")
+def basis_dtype(request):
+    """
+    Dtypes for basis
+    """
+    return request.param
+
+
+@pytest.fixture(params=IMG_SIZE, ids=lambda x: f"count={x}", scope="module")
+def img_size(request):
+    """
+    Image size for basis.
+    """
+    return request.param
+
+
+@pytest.fixture(params=STACKS, ids=lambda x: f"stack={x}", scope="module")
+def stack(request):
+    """
+    Stack dimensions.
+    """
+    return request.param
+
+
+@pytest.fixture(params=ALLYOURBASES, ids=lambda x: f"basis={x}", scope="module")
+def basis(request, img_size, basis_dtype):
+    """
+    Parameterized `Basis` instantiation.
+    """
+    cls = request.param
+    return cls(img_size, dtype=basis_dtype)
+
+
+@pytest.fixture(scope="module")
+def coef_fixture(basis, stack, dtype):
+    """
+    Construct parameterized testing coefficient array as `Coef`.
+    """
+    # Combine the stack and coeficent counts into multidimensional
+    # shape.
+    size = stack + (basis.count,)
+
+    coef_np = np.random.random(size=size).astype(dtype, copy=False)
+
+    return Coef(basis, coef_np, dtype=dtype)
+
+
+@pytest.fixture(scope="module")
```

Comment:
Nice. I need to take advantage of `scope="module"` in more of my tests.

Reply:
These particular tests would take way too long without it.